rotkehlchen/accounting/cost_basis/base.py

```diff
@@ -136,42 +156,268 @@ def consume_result(self, used_amount: FVal) -> None:
+        self._count += 1
+
+    def consume_result(self, used_amount: FVal) -> None:
+        assert ZERO <= used_amount <= self._acquisitions[0][1].remaining_amount, f'Used amount must be in the interval [0, {self._acquisitions[0][1].remaining_amount}] but it was {used_amount}'  # noqa: E501
+
+        self._acquisitions[0][1].remaining_amount -= used_amount
+        self.remaining_amount -= used_amount
+        if self._acquisitions[0][1].remaining_amount == ZERO:
+            heapq.heappop(self._acquisitions)
+
+    def calculate_spend_cost_basis(
+            self,
+            spending_amount: FVal,
+            spending_asset: Asset,
+            timestamp: Timestamp,
+            missing_acquisitions: List[MissingAcquisition],
+            used_acquisitions: List[AssetAcquisitionEvent],  # pylint: disable=unused-argument
+            settings: DBSettings,
+            timestamp_to_date: Callable[[Timestamp], str],  # pylint: disable=unused-argument
+            average_cost_base: Optional[FVal] = None,  # pylint: disable=unused-argument
+    ) -> 'CostBasisInfo':
+        """
+        Calculates the cost basis of the spend using the average cost base method.
+
+        Formula is taken from:
+        https://www.adjustedcostbase.ca/blog/how-to-calculate-adjusted-cost-base-acb-and-capital-gains/
+        """
+        if self.remaining_amount == ZERO:
+            missing_acquisitions.append(
+                MissingAcquisition(
+                    asset=spending_asset,
+                    time=timestamp,
+                    found_amount=self.remaining_amount,
+                    missing_amount=spending_amount,
+                ),
+            )
+
+            return CostBasisInfo(
+                taxable_amount=ZERO,
+                taxable_bought_cost=ZERO,
+                taxfree_bought_cost=ZERO,
+                matched_acquisitions=[],
+                is_complete=False,
+            )
+
+        self.current_average_cost_base *= ((self.remaining_amount - spending_amount) / self.remaining_amount)  # noqa: E501
+        return super().calculate_spend_cost_basis(
+            spending_amount=spending_amount,
+            spending_asset=spending_asset,
+            timestamp=timestamp,
+            missing_acquisitions=missing_acquisitions,
+            used_acquisitions=used_acquisitions,
+            settings=settings,
+            timestamp_to_date=timestamp_to_date,
+            average_cost_base=self.current_average_cost_base,
+        )
 
 
 class CostBasisEvents:
     used_acquisitions: List[AssetAcquisitionEvent]
-    acquisitions_manager: BaseAcquisitionsOrder
+    acquisitions_manager: BaseCostBasisMethod
     spends: List[AssetSpendEvent]
 
     def __init__(self, cost_basis_method: CostBasisMethod) -> None:
-        """This class contains data about acquisitions and spends. `acquisitions` field contains
+        """
+        This class contains data about acquisitions and spends. `acquisitions` field contains
```

Comment:
Is there an acquisitions attribute? I don't see one. Why mention it in the docstring?